clippy_lints/src/utils/conf.rs

```diff
@@ -402,6 +402,10 @@ define_Conf! {
     /// A list of paths to types that should be treated like `Arc`, i.e. ignored but
     /// for the generic parameters for determining interior mutability
     (ignore_interior_mutability: Vec<String> = Vec::from(["bytes::Bytes".into()])),
+    /// Lint: INDEXING_SLICING
+    ///
+    /// Whether to suppress lint in const function
```

Comment:
I think here it's important to document, why a user might want/not want to enable this. Presently, it sounds like there is no reason to set it to true.

```suggestion
    /// Whether to suppress a restriction lint in constant code. In same
    /// cases the restructed operation might not be unavoidable, as the
    /// suggested counterparts are unavailable in constant code. This
    /// configuration will cause restriction lints to trigger even
    /// if no suggestion can be made.
```

Something like that, I feel like it's not perfect yet, but good enough ^^